### ISSUE TITLE
Spherical_sizing_field requires FT member?

### ISSUE THREAD

**Issue by nschloe** [CGAL/cgal#4146]
When working with sizing fields (as in https://doc.cgal.org/latest/Mesh_3/index.html#title27), I notice that things only work if the `struct`, i.e., `struct Spherical_sizing_field`, has an attribute `FT` defined. The code in the above link does that by taking `FT` from the global scope,
```c++
struct Spherical_sizing_field
{
  typedef ::FT FT;   // <----  !!!
  typedef Point Point_3;
  typedef Mesh_domain::Index Index;
  
  FT operator()(const Point_3& p, const int, const Index&) const
  {
    FT sq_d_to_origin = CGAL::squared_distance(p, Point(CGAL::ORIGIN));
    return CGAL::abs( CGAL::sqrt(sq_d_to_origin)-0.5 ) / 5. + 0.025; 
  }
};
```
If it wasn't for the that, the field could simply be a function.

Wouldn't it be possible to lift this requirement?

**Comment by afabri** [comment]
Technically no code uses the nested `typedefs`, so the following works:

```
struct Spherical_sizing_field
{
  double operator()(const Point& p, const int, const Mesh_domain::Index&) const
  {
    double sq_d_to_origin = CGAL::squared_distance(p, Point(CGAL::ORIGIN));
    return CGAL::abs( CGAL::sqrt(sq_d_to_origin)-0.5 ) / 5. + 0.025; 
  }
};
````

**Comment by lrineau** [comment]
That issue has been address by the PR https://github.com/CGAL/cgal/pull/3953. The code is available in the `master` branch of the CGAL Git repository. It will be published in CGAL-5.0, in Autumn 2019.

**Comment by afabri** [comment]
@nschloe  when I write that the version without typedefs compiles I didn't realize that I tried this in master, so it maybe does not work with 4.14.
